### ISSUE TITLE
`PYTHONPATH` not being setup via `pbench-trampoline`

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#2418]
We appear to have a problem since we don't update the `PYTHONPATH` to include the various `/opt/pbench-server` Python3 library paths we have (for the python modules the pbench-server code provides, and for the local install of `pip3` dependent modules for the given version of Python3 being used).

We need to properly fix the missing Python3 `PYTHONLIB` paths IFF we want to support Fedora 34 environments for our pbench-server installs.

_Originally posted by @portante in https://github.com/distributed-system-analysis/pbench/issues/2415#issuecomment-900793845_

**Comment by dbutenhof** [comment]
I'm not sure this is still relevant with the switch to `--user` install, especially given that our service and crontab already run as `pbench` user (which will now see our `/opt/pbench-server/lib` install path and `requirements.txt` installs by default without a `PTYHONPATH` at all.)

Should this be closed, or should we keep it open as a hook to investigate and address the implied Fedora packaging requirements in the future? (E.g., if we need to move entirely to RPM installs, which is probably desirable, much of the literal text here goes away but there's a core placeholder that may be worth keeping?)

**Comment by webbnh** [comment]
This comment is (mostly) just for my own edification, shared here in case it's of any use to anyone else.

We have three scenarios in which we run Python-based commands:

1. Agent-side commands
1. Server-side deployment and background activities
1. New, server-side administrative commands

In each case, we need to ensure that the "user" running the command can get to our Python modules.

In a deployment, we have reasonable solutions, at least for the moment.  In the first case, we've have a shell initialization script (`agent/profile`) which we expect the user to run as part of login, etc., which adds the relevant paths to the user's `PYTHONPATH`.  In the second case, we "now" install the modules in a place where Python will look for them implicitly, so, as long as the tools are invoked with the intended UID, no modification of the `PYTHONPATH` is necessary.  For the third case, we don't really have a solution, yet, but I expect that it will follow the Agent-side (although, I don't know how robust/resilient this approach is -- we might want to replace it with another approach, now that we're more familiar with "site packages").

For testing, the situation is less clear-cut.  For the first case, the test framework _emulates_ `agent/profile`, which is possibly a very bad idea.  For the second case, I believe that we're currently taking a similar approach, but, hopefully this will change under the new testing regime (where the test set-up will actually perform an installation, and so the `--user` switch should do its magic; perhaps this will enable an improvement to the first case, as well).  For the third case, once again, I presume it will follow whatever we do for the first case (I'm not sure that we're actually testing these commands, currently).

So, in conclusion (if you've read this far...), I think that this issue has been resolved so far as the Server's own activities are concerned (or, in the case of testing, it _will_ be resolved, soon, under separate cover).  So, in that sense, I think this issue can be closed.  We do have lingering concerns around how the user commands for the Agent and Server are invoked, particularly for testing purposes, but I think those are outside the scope of the original intent of this issue (and, they may indeed be represented in separate issues and/or requirements).

**Comment by dbutenhof** [comment]
For your case 3, the "official answer" (that is, Peter proposed it, and I think it's not a bad idea) is that the server Pbench administrative commands (pbench-user-create, etc.) are *only* supported as the `pbench` user, where the `PYTHONPATH` is implicit.

This isn't particularly onerous, since as root it's trivial to just prefix with `su pbench -c` and you're good to go.

**Comment by portante** [comment]
Won't fix this, see previous comments for reasoning.
